Supporting a scenario where proto_library is being used withour srcs...

amitz opened the pull request:
#### References to other Issues or PRs ####
Fixes #2208

#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/CONTRIBUTING.md)?
Yes

#### Brief description of what is fixed or changed
Instead of using `proto_source_root`, `check_deps_sources.to_list()`, which return the source if exists, and if not - support the other scenario of a `proto_library` without srcs and only deps.

achew22 commented:
Your commit message says that you've added an example, but I can't see it in the diff. Can you help me understand what's going on?

amitz commented:
Your right, sorry. I originally added an example of the "Public API" scenario (detailed [here](https://github.com/grpc-ecosystem/grpc-gateway/issues/2208)) into `examples/internal/proto/examplepb/BUILD.bazel`, but I removed it during the re-base and forgot to edit the description. 

The reason I removed was that in after thought - it made the Example file more confusing. I was trying to emphasis a `proto_library` concept (of using just `deps`) with the current protobufs available at the example folder, and it didn't make a lot of sense. I ended up decided using `deps` instead of just `srcs` is more of a `proto_library` logic, and doesn't directly intertwined into how to use `protoc_gen_openapiv2`. 

Let me know if you think otherwise. 
Thanks.

achew22 commented:
I agree that 1.0 compatibility is overrated and not worth persisting. Just wanted to check that there wasn't something missing.
